src/Symfony/Component/Yaml/Parser.php

```diff
@@ -730,15 +730,15 @@ private function cleanup($value)
         $this->offset += $count;
 
         // remove leading comments
-        $trimmedValue = preg_replace('#^(\#.*?\n)+#s', '', $value, -1, $count);
+        $trimmedValue = preg_replace('#^\n*\s*(\#.*?\n)+#s', '', $value, -1, $count);
         if ($count == 1) {
             // items have been removed, update the offset
             $this->offset += substr_count($value, "\n") - substr_count($trimmedValue, "\n");
             $value = $trimmedValue;
         }
 
         // remove start of the document marker (---)
-        $trimmedValue = preg_replace('#^\-\-\-.*?\n#s', '', $value, -1, $count);
+        $trimmedValue = preg_replace('#^\n*\s*\-\-\-.*?\n#s', '', $value, -1, $count);
```

Comment:
@BR0kEN- http://yaml-online-parser.appspot.com/, for example, doesn't allow spaces before the document marker. And having spaces before `---` clearly means that the document marker is not at the beginning of the line (there is nothing in the specs indicating that you could simply ignore these spaces).